### ISSUE TITLE
[FlutterFire_UI]: SignInScreen blocked by keyboard on small sized mobile devices

### ISSUE THREAD

**Issue by fredecodes** [firebase/flutterfire#8084]
Simple to reproduce this:
1) Create a SigninScreen widget with an image that takes up only the top 25% of the screen
2) Attempt to use it on any medium or small sized Android/iOS device.
3) The password field will be blocked by the keyboard.  You can still type a password, but won't be able to see your typing.  

Please make the contents of this screen scrollable so that the user can scroll within the remaining screen space above the keyboard.

### Discussed in https://github.com/FirebaseExtended/flutterfire/discussions/8050

<div type='discussions-op-text'>

<sup>Originally posted by **fredecodes** February  9, 2022</sup>
There seems to be a really big problem with SigninScreen that I've not seen mentioned elsewhere, which surprises me as I see it as a huge usability issue.  When rendered on a regular or small sized mobile screen, the "Password" text field is completely hidden by the keyboard (on both Android and iOS).  You can still type to enter text, but you are typing in the blind because you can't see the cursor.  This is very frustrating for users, especially when it's their very first impression of your app.  When the keyboard is up, the screen should scroll to fit within the upper bounds that are visible.  Am I missing something, is there a way around this to let the user see the text field that is hidden by the keyboard?</div>

**Comment by darshankawar** [comment]
I ran `flutterfirre_ui` plugin example on couple of simulators and devices, but they are not small sized, so I am unable to replicate this behavior. The closest I could verify is below, but there, it seems to be fine:

<img width="742" alt="Screenshot 2022-02-15 at 4 26 46 PM" src="https://user-images.githubusercontent.com/67046386/154048339-81fec3c9-682f-4d4f-a2b7-d79911757d55.png">


Labeling it for further insights from the team.

/cc @lesnitsky

**Comment by fredecodes** [comment]
Thanks for your response.  When I run the sample plugin I see the same result as you -- it works fine.  But if you make the following 3 changes to the example code, you will see the bug:
1) Change providerConfigs to use only EmailProviderConfiguration
2) Comment out the footerBuilder
3) Change the subtitleBuilder text to be 3 lines instead of 2 (I just copied the existing text twice). 

Here is a screenshot from the iOS simulator using "iPhone SE - Second generation" device:
![Simulator Screen Shot - iPhone SE (2nd generation) - 2022-02-15 at 13 19 04](https://user-images.githubusercontent.com/15000050/154151756-7955809c-c0bb-44e8-ad72-a6c2299972a0.png)

At the time the screenshot was taken, I am entering the password, but am not able to scroll the screen any lower, so I can only see the top half of the password text field and have no way to see the letters I am typing.

It seems like as long as there is enough content below the "Sign In" button, the scrolling works, so as a temporary workaround I added some blank footer text which occupies several lines.  This works but is not a real fix.
